server/pipeline/items.go

```diff
@@ -60,6 +60,10 @@ func createBuildItems(ctx context.Context, store store.Store, build *model.Build
 		}
 	}
 
+	for k, v := range build.Variables {
+		envs[k] = v
```

Comment:
This can overwrite all existing one, is this intemded behaviour?

Reply:
I would say it should not be able to override existing vars as that could be a security risk as well or we should think about giving them a prefix.

Reply:
How can you abuse it?

Reply:
You could bypass some filter based on env vars, but I noticed that as you need push access to trigger a manual pipeline this is not a problem.